Tests/ApolloInternalTestHelpers/MockURLProtocol.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+
+public class MockURLProtocol: URLProtocol {
+  
+  public typealias MockRequestHandler = ((URLRequest) throws -> Result<(HTTPURLResponse, Data?), Error>)
+  
+  // Dictionary of mock request handlers where the `key` is the URL of the request.
+  public static var requestHandlers = [URL: MockRequestHandler]()
```

Comment:
When unit tests are mutating static properties, we should use a `TestObserver` to ensure that state is reset between tests. Especially for tests that run async code, you can easily break test isolation.

Check out the `MockSchemaMetadata` to see an example of using `TestObserver`.

Actually, since we run unit tests in parallel, this could still be an issue even with the `TestObserver`. Is there a way to indicate that certain tests must not be run in parallel I wonder?